### PR TITLE
add hf/gpt2 test

### PR DESCRIPTION
**Pull request by suexu1025**
# Description

Please include a summary of relevant context/issue and your changes.
Added hf/pt/GPT2 test on V4-8

# Tests
tested for 
pt-nightly-hf-gpt2-clm-2b-pjrt-conv-v4-8-1vm on v4-8
**List links for your tests (use go/shortn-gen for any internal link):** ...
"http://shortn/_PbhqATHDdc"
# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ x] I have performed a self-review of my code.
- [ x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ x ] I have run one-shot tests and provided workload links above if applicable. 
- [ x ] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: tests/pytorch/nightly/hf-llm.libsonnet [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    accelerator: tpus.v4_8,
+  },
+
+  local pjrt = self.pjrt,
+  pjrt:: tpuVm + experimental.PjRt {
+    modelName+: '-pjrt',
+  },
+
+  configs: [
+    gpt2_model + v4_8 + config_2B + timeouts.Hours(1) + pjrt,
```
Comment (RissyRan) [inline tests/pytorch/nightly/hf-llm.libsonnet:145]: What's the default timeout in PjRt?

File: tests/pytorch/nightly/hf-llm.libsonnet [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        export LD_LIBRARY_PATH=/usr/local/lib/
+        export PJRT_DEVICE=TPU_C_API
+        export PT_XLA_DEBUG=0
+        export USE_TORCH=ON
+      |||,
+      tpuVmExtraSetup: |||
+        pip install --upgrade accelerate
+        git clone https://github.com/huggingface/transformers.git
+        cd transformers
+        git checkout ebdb185befaa821304d461ed6aa20a17e4dc3aa2
```
Comment (RissyRan) [inline tests/pytorch/nightly/hf-llm.libsonnet:124]: May I know the reason for that specific commit?

File: tests/pytorch/nightly/hf-llm.libsonnet (hunk 1 above)
Comment (suexu1025) [reply tests/pytorch/nightly/hf-llm.libsonnet:145]: it doesn't change the default timeout setting

File: tests/pytorch/nightly/hf-llm.libsonnet (hunk 2 above)
Comment (suexu1025) [reply tests/pytorch/nightly/hf-llm.libsonnet:124]: the latest commit will have crash, so this is the stable version we can use.

File: tests/pytorch/nightly/hf-llm.libsonnet (hunk 1 above)
Comment (RissyRan) [reply tests/pytorch/nightly/hf-llm.libsonnet:145]: I remember the functional test limit is 1 hour, so wondering why we have `timeouts.Hours(1)` here? Or this is a convergence test?

**Review by RissyRan — COMMENTED** [review]
Overall, it looks good me. Just have one follow-up question about the time-out.

File: tests/pytorch/nightly/hf-llm.libsonnet (hunk 1 above)
Comment (suexu1025) [reply tests/pytorch/nightly/hf-llm.libsonnet:145]: It is convergence test, and mostly it ends in 30min for 3 epochs. So 1 hour timer here is to monitor this test's performance regression.

Comment (RissyRan) [reply tests/pytorch/nightly/hf-llm.libsonnet:145]: Thanks! This sounds good to me.

**Review by RissyRan — APPROVED** [review]

**Pull request merged**